File: 20221202/Bracket.test.js
import solution from './Bracket';

test('Bracket', () => {
  expect(solution('()()')).toBe(true);
  expect(solution('(())()')).toBe(true);
  expect(solution(')()(')).toBe(false);
  expect(solution('(()(')).toBe(false);
  expect(solution('())(()')).toBe(false);
});

File: 20221202/Bracket/src/test/java/BracketTest.java
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class BracketTest {
    @Test
    void solution() {
        Bracket test = new Bracket();

        assertTrue(test.solution("()()"));
        assertTrue(test.solution("()()"));
        assertFalse(test.solution(")()("));
        assertFalse(test.solution("(()("));
        assertFalse(test.solution("(()("));
        assertFalse(test.solution("())(()"));
    }
}
